From c13f785cc9ff321a02a6a1e4a1c732c004e894b2 Mon Sep 17 00:00:00 2001
From: Alessandro Decina <alessandro.decina@collabora.co.uk>
Date: Mon, 13 Jun 2011 02:49:37 +0200
Subject: [PATCH 013/119] camerabin2: swap image-filter and image-capture-caps

Make it so that the capsfilter controlled by image-capture-caps is linked before
the image-filter element. Makes it more consistent in that image-capture-caps
always filters the output of the camera and can so be used to negotiate with the
camera.
---
 gst/camerabin2/gstcamerabin2.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/gst/camerabin2/gstcamerabin2.c b/gst/camerabin2/gstcamerabin2.c
index f703798..ce53ebc 100644
--- a/gst/camerabin2/gstcamerabin2.c
+++ b/gst/camerabin2/gstcamerabin2.c
@@ -1141,8 +1141,8 @@ gst_camera_bin_create_elements (GstCameraBin * camera)
   }
 
   gst_camera_bin_check_and_replace_filter (camera, &camera->image_filter,
-      camera->user_image_filter, camera->imagebin_queue,
-      camera->imagebin_capsfilter);
+      camera->user_image_filter, camera->imagebin_capsfilter,
+      camera->image_encodebin);
   gst_camera_bin_check_and_replace_filter (camera, &camera->video_filter,
       camera->user_video_filter, camera->videobin_queue,
       camera->videobin_capsfilter);
-- 
1.7.5.4

